tools/launchdev.sh

```diff
@@ -415,8 +414,24 @@ function st2start(){
             --config-file $ST2_CONF --register-all
     fi
 
-    if [ "$copy_examples" = true ]; then
-        st2 run packs.setup_virtualenv packs=tests
+    if [ "$copy_test_packs" = true ]; then
+        # Check if authentication is enabled
+        while read -r line; do
+            if [ "$line" == "[auth]" ]; then
+                found_auth=true
+            fi
+
+            if [ "$found_auth" = true ]; then
+                typeset -l line
+                if [ "$line" = "enable = true" ]; then
+                    echo "Warning: Please setup virtualenv for pack \"tests\" before run integration test"
+                    break
+                elif [ "$line" = "enable = false" ]; then
+                    st2 run packs.setup_virtualenv packs=tests
+                    break
+                fi
+            fi
+        done < "$ST2_CONF"
```

Comment:
Instead of this while loop, can we just run `st2 run packs.setup_virtualenv packs=tests` and if it returns `ERROR: 401 Client Error: Unauthorized`, you echo the warning?

Reply:
I thought `just run st2 run packs.setup_virtualenv packs=tests and if it returns ERROR: 401 Client Error: Unauthorized` is easy way but not graceful way.

Reply:
It's ok with me.

Reply:
Are you going to change this?

Reply:
> Are you going to change this?

I am not going to change, but if this is too much, i can change it.

Reply:
Yeah, I prefer simplicity here.  If it returns an auth error, then warn users.  This is a dev launch script, don't need to go thru the hassle to parse for the auth lines in the config file.